AWSCognitoAuth/AWSCognitoAuth.m

```diff
@@ -354,6 +354,9 @@ - (void) completeGetSession: (nullable AWSCognitoAuthUserSession *) userSession
     if(error){
         self.getSessionError = error;
         [self.getSessionQueue cancelAllOperations];
+        if(self.getSessionBlock){
+           self. getSessionBlock(nil, error);
```

Comment:
The problem I was seeing is that `cleanUpAndCallGetSessionBlock:error` does not get called. It enters the `dismissSafariVC:` method but `dismissViewControllerAnimated:NO completion:`'s completion block never gets executed for some reason. Maybe a better PR would be putting the `cleanUpAndCallGetSessionBlock:error` outside of the `dismissSafariVC:`'s completion block?

Reply:
I see, we have seen some other customer's report, for e.g. #1043 where the callback might not get called in some cases. 

I think it might have to do with the safari view controller being set to nil before calling the completion handler. Refer https://github.com/aws-amplify/aws-sdk-ios/blob/master/AWSCognitoAuth/AWSCognitoAuth.m#L681-L682 

I think the fix should be to call the block before de-allocating the safari view controller. Do you agree with the change? Or do you think it is a better idea to call it outside of dismiss scope?

Reply:
I updated my PR to call it outside of the dismiss scope. I can test your idea and see if call the block before setting the svc to nil will fix the issue. If that works I will update my PR again.

Reply:
Awesome, thanks @zhentan 

I will wait for an update from you. Will review these changes in the meanwhile.

Reply:
Changing the call order fixed the issue as you have predicted! PR updated.